Framework/DataHandling/src/SaveGSS.cpp

```diff
@@ -92,391 +136,502 @@ void SaveGSS::init() {
+  *
+  * @param outFilePath :: The full path of the file to open a stream out
+  * @throws :: If the fail bit is set on the out stream. Additionally
+  * logs the system reported error as a Mantid error.
+  * @return :: The opened file stream at the user specified path
   */
-inline double fixErrorValue(const double value) {
-  if (value <= 0. ||
-      !std::isfinite(value)) // Negative errors cannot be read by GSAS
-    return 0.;
-  else
-    return value;
+std::ofstream SaveGSS::openFileStream(const std::string &outFilePath) {
+  const bool append = getProperty("Append");
+
+  // Select to append to current stream or override
+  using std::ios_base;
+  const ios_base::openmode mode = (append ? (ios_base::out | ios_base::app)
+                                          : (ios_base::out | ios_base::trunc));
+
+  auto outStream = std::ofstream(outFilePath, mode);
+  if (outStream.fail()) {
+    // Get the error message from library and log before throwing
+    const std::string error = strerror(errno);
+    g_log.error("Failed to open file. Error was: " + error);
+    throw std::runtime_error("Could not open the file at the following path: " +
+                             outFilePath);
+  }
+  // Stream is good - return to caller through a std::move
+  return std::move(outStream);
+}
+
+/** Ensures that when a workspace group is passed as output to this workspace
+*  everything is saved to one file and the bank number increments for each
+*  group member.
+*  @param alg ::           Pointer to the algorithm
+*  @param propertyName ::  Name of the property
+*  @param propertyValue :: Value  of the property
+*  @param periodNum ::     Effectively a counter through the group members
+*/
+void SaveGSS::setOtherProperties(IAlgorithm *alg,
+                                 const std::string &propertyName,
+                                 const std::string &propertyValue,
+                                 int periodNum) {
+  // We want to append subsequent group members to the first one
+  if (propertyName == "Append") {
+    if (periodNum != 1) {
+      alg->setPropertyValue(propertyName, "1");
+    } else
+      alg->setPropertyValue(propertyName, propertyValue);
+  }
+  // We want the bank number to increment for each member of the group
+  else if (propertyName == "Bank") {
+    alg->setProperty("Bank", std::stoi(propertyValue) + periodNum - 1);
+  } else
+    Algorithm::setOtherProperties(alg, propertyName, propertyValue, periodNum);
 }
 
 /**
+  * Validates the user input is matches certain constraints
+  * in length and type and logs a warning or throws depending on
+  * whether we can continue.
+  *
+  * @throws:: If for any reason we cannot run the algorithm
   */
-void SaveGSS::writeRALFdata(const int bank, const bool MultiplyByBinWidth,
-                            std::stringstream &out,
-                            const HistogramData::Histogram &histo) const {
+void SaveGSS::validateUserInput() const {
```

Comment:
I had called the base class `ValidateInputs` rather than this function - good catch